Chewbawka2017/src/org/team1540/chewbawka/Shooter.java

```diff
@@ -1,7 +1,98 @@
+				flywheelShootingVelocity, // spinup
+				flywheelShootingVelocity, // firing
+				flywheelCompensateVelocity); // compensate
+
+		// Set the speed of the intake
+		FloatInput intakeShootingSpeed = Robot.mainTuning.getFloat("Shooter Intake Speed", 1f);
+
+		FloatInput intakeSpeed = shooterStates.selectByState(
+				FloatInput.zero, // passive
+				FloatInput.zero, // spinup
+				intakeShootingSpeed, // firing
+				FloatInput.zero); // compensate
+
+		// Set the target velocity of the belt
+		FloatInput beltTargetShootingVelocity = Robot.mainTuning.getFloat("Shooter Belt Target Velocity", 1000f);
+
+		FloatInput beltTargetVelocity = shooterStates.selectByState(
+				FloatInput.zero, // passive
+				FloatInput.zero, // spinup
+				beltTargetShootingVelocity, // firing
+				FloatInput.zero); // compensate
+
+		// Control bindings
+		ControlBindings.fireButton.onRelease().send(shooterStates.getStateSetEvent("passive"));
+
+		// Setup flywheel PID controller
+		
+		flywheelLeft.modGeneralConfig().configureReversed(true, true);
+		flywheelLeft.modGeneralConfig().activateFollowerMode(flywheelRight);
+
+		PIDTalon flywheelTalon = new PIDTalon(flywheelRight, "Shooter Flywheel", flywheelTargetVelocity);
+		flywheelTalon.setup();
+		
+		// Setup belt PID controller 
+		PIDTalon beltTalon = new PIDTalon(shooterBelt, "Shooter Belt", beltTargetVelocity);
+		beltTalon.setup();
+
+		// Set tunable variables
+		FloatInput shooterSlowThreshold = Robot.mainTuning.getFloat("Shooter Slowdown Threshold", 2100f);
+
+		// Start switching logic
+		ControlBindings.fireButton.onPress().and(shooterStates.getIsState("passive")).send(shooterStates.getStateSetEvent("spinup"));
+		flywheelTalon.isUpToSpeed.onPress().and(shooterStates.getIsState("spinup")).send(shooterStates.getStateSetEvent("firing"));
+		flywheelTalon.velocity.atMost(shooterSlowThreshold).onPress().and(shooterStates.getIsState("firing")).send(shooterStates.getStateSetEvent("compensate"));
+		flywheelTalon.velocity.atLeast(flywheelShootingVelocity).onPress().and(shooterStates.getIsState("compensate")).send(shooterStates.getStateSetEvent("firing"));
+
+		// Reset the state machine when switching modes
+		shooterStates.setStateWhen("passive", FRC.startDisabled.or(FRC.startTele).or(FRC.startAuto).or(FRC.startTest));
+
+		// Send intake speeds
+		FloatInput shooterFrontConveyorConstant = Robot.mainTuning.getFloat("Shooter Front Conveyor Constant", .5f);
+		FloatInput shooterFunnelingRollerConstant = Robot.mainTuning.getFloat("Shooter Funneling Roller Constant", 1f);
+		
+		FloatOutput shooterFunnelingRoller = shooterFunnelingRollerLeft.simpleControl().combine(shooterFunnelingRollerRight.simpleControl().negate());
+		
+		intakeSpeed.multipliedBy(shooterFrontConveyorConstant).send(shooterFrontConveyor.simpleControl().addRamping(.02f, FRC.constantPeriodic));
+		intakeSpeed.multipliedBy(shooterFunnelingRollerConstant).send(shooterFunnelingRoller.addRamping(.02f, FRC.constantPeriodic));
```

Comment:
Need to have something so the balls keep moving.

- Could have 1 on at a constant speed and the other pulsing on/off (fab’s suggestion)
- Could have both on constantly but at different speeds (Jake’s suggestion)